src/test/ui/suggestions/issue-96223.rs

```diff
@@ -0,0 +1,53 @@
+// Test case for #96223.
+// An ICE was triggered because of a failed assertion.
+// Thanks to @Manishearth for the minimal test case.
```

Comment:
I think this comment can be left out.

Can you also change this to be `// check-pass`? We don't need to actually build it.